Sources/TextBufferKit/PieceTreeBase.swift

```diff
@@ -1150,10 +1150,10 @@ public class PieceTreeBase {
                 if lastChar == 13 || (lastChar >= 0xD800 && lastChar <= 0xDBFF) {
                     // last character is \r or a high surrogate => keep it back
                     splitText = Array (text [0..<(AverageBufferSize - 1)])
-                    text = text [(AverageBufferSize - 1)...]
+                    text = Array(text [(AverageBufferSize - 1)...])
                 } else {
                     splitText = Array (text [0..<AverageBufferSize])
-                    text = text[AverageBufferSize...]
```

Comment:
here, due to the nature of ArraySlice, the new array slice doesn't start at index `0`, but at `text.startIndex`, meaning everything has to be adjusted to the slice range.